source/slang/slang-ast-dump.cpp

```diff
@@ -253,7 +249,7 @@ struct ASTDumpContext
         buf.appendChar('\"');
         for (const char c : slice)
         {
-            if (_isPrintableChar(c))
+            if (std::isprint(c))
```

Comment:
Err. std::isprint behavior is different than _isPrintableCharacter (it uses locale for starters). It's not clear to me that we want this change. https://en.cppreference.com/w/cpp/string/byte/isprint

Reply:
ah yes, that was lazy on my part not checking the context properly (I assumed that if we're "dumping" something we'll want it locale compatible). I'll change that one back

Reply:
pissed that gcc warns on this:

```
../../../source/slang/slang-ast-dump.cpp:247:56: warning: comparison is always true due to limited range of data type [-Wtype-limits]
  247 |         static_assert(std::numeric_limits<char>::max() <= 0x80);
      |                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
```

Reply:
However, I think this is a potential, albeit unlikely bug

Consider the case of dumping a string "\DEL". One could imagine a handling of \DEL here which leads to the open quote symbol being deleted.

I've changed things to be a predicate over needing escaping instead, which conveniently makes this an important check.